packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -145,8 +145,8 @@ function mark_reactions(signal, status) {
 		var reaction = reactions[i];
 		var flags = reaction.f;
 
-		// Skip any effects that are already dirty
-		if ((flags & DIRTY) !== 0) continue;
+		// Skip if they're already dirty or if they're maybe dirty and not unowned
+		if ((flags & DIRTY) !== 0 || ((flags & MAYBE_DIRTY) !== 0 && (flags & UNOWNED) === 0)) continue;
```

Comment:
This logic is too convoluted for my simple brain. 'skip' and 'not' and 'un' is three negatives in combination — by the end of it I have no idea what's going on!

If I change it to this, all the tests continue to pass. Is it equivalent?

```suggestion
		// If a reaction is already dirty, skip it (but always mark unowned deriveds)
		if ((flags & (CLEAN | UNOWNED)) === 0) continue;
```

Reply:
Actually, it's not right. This ironically breaks https://github.com/sveltejs/svelte/pull/12545